src/modules/timepix3/generators/build_tof_test_tpx.py

```diff
@@ -0,0 +1,153 @@
+# Copyright (C) 2024 European Spallation Source, ERIC. See LICENSE file
+# ===----------------------------------------------------------------------===#
+#
+# Brief: This code is generating a tpx file which can be used with the tpx
+#        generator. The tpx generator is a tool that generates udp packets
+#        from tpx data and sends it to the EFU.
+#
+#        This code is generating tpx file with a minimal data to initialize
+#        the timepix efu and send some pixel packets with exact delays to
+#        test TOF calculation.
+# ===----------------------------------------------------------------------===#
+
+import bitstruct
+import struct
+import numpy as np
+import sys
+import argparse
+
+PIXEL_MAX_TIMESTAMP_NS = 26843545600
+UDP_MAX_SIZE = 8952
+
+
+def gen_tdc_readout(counter, tdc_ns):
+    # Define the tdc format of the payload
+    format = "u4u4u12u35u4u5"
```

Comment:
Might be good with an explanation. Either a reference to the data format in a document or 
a brief text.